web-app/HeimScripts/heatmap/run.R

```diff
@@ -1,187 +1,254 @@
 library(reshape2)
 
-
-main <- function(max_rows=100, sorting="nodes"){
+main <- function(max_rows = 100, sorting = "nodes") {
   max_rows <- as.numeric(max_rows)
   verifyInput(max_rows, sorting)
-  if(exists("preprocessed")){
-    df <- preprocessed
-  }
-  else{
-    df <- mergeFetchedData(loaded_variables)
+  df <- parseInput()
+  df["Row.Label"] <-
+    lapply(df["Row.Label"],fixString)  # remove illegal
+                                       # characters from probe names. This will
+                                       # prevent problems with CSS
+                                       # selectors on the frontend.
+  write.table(
+      df,
+      "heatmap_orig_values.tsv",
+      sep = "\t",
+      na = "",
+      row.names = FALSE,
+      col.names = TRUE
+    )
+  df <- addStats(df, sorting)
+  df <- df[1:min(max_rows,nrow(df)),]  #  apply max_rows
+  fields <- buildFields(df)
+  extraFields <- buildExtraFields(fields)
+  probes <- na.omit(df[,1])
+  geneSymbols <- df[,2][1:length(probes)]
+  #unique(fields["GENESYMBOL"])[,1] #[,1] in order to get a vector,
+  # otherwise we get a dataframe
+  patientIDs <- unique(fields["PATIENTID"])[,1]
```

Comment:
can also be done like this unique(fields[["PATIENTID"]]) or unique(fields[,"PATIENTID"]). I usually use the second option as it is most intuitive for me to treat data.frame as a table with data.frame[rows, columns]. In case of first alternative you treat it as a list (which it also is to R)